cmd/adapter-rest/startcmd/start.go

```diff
@@ -271,9 +280,20 @@ func addRPHandlers(parameters *adapterRestParameters, router *mux.Router) error
 		return err
 	}
 
+	datasource, err := initDB(parameters.dsn)
+	if err != nil {
+		return err
+	}
+
+	// TODO init OIDC stuff in iteration 2
+
+	// add rp endpoints
 	rpService, err := rp.New(&operation.Config{
 		PresentationExProvider: presentationExProvider,
 		Hydra:                  newHydraClient(hydraURL).Admin,
+		TrxProvider:            newTrxProvider(datasource),
+		UsersDAO:               &db.EndUsers{DB: datasource},
```

Comment:
It would be better to init with a constructor than direct struct initiation.

Reply:
done